### ISSUE TITLE
[Feature] Enable code formatting, go imports, linters in CI

### ISSUE THREAD

**Issue by Jeffwan** [ray-project/kuberay#76]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/kuberay/issues) and found no similar feature requirement.

### Description

We may not always remember to do this work which results in inconsistent formatting issue. 
Let's create the standard to help us maintain the code here.

- gofmt
- goimports
- golangci-lint

```
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.23.7

golangci-lint run  ./... # introduce linter config.yaml if necessary
```

I think the overall idea is to improve project quality.  See suggestions here https://goreportcard.com/report/github.com/ray-project/kuberay

![image](https://user-images.githubusercontent.com/4739316/137640774-55d45012-e386-4852-8ea1-d72723aea395.png)

![image](https://user-images.githubusercontent.com/4739316/137640780-54ecd14b-bdec-4b85-be97-e1a3a2db1cd9.png)

### Use case

To avoid inconsistency issue from different contributors.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by wilsonwang371** [comment]
Is this bug done?

**Comment by Jeffwan** [comment]
@wilsonwang371 which bug?

**Comment by Jeffwan** [comment]
golint has been marked as deprecated in golangci-lint. Above screenshot is not that helpful anymore. Default linter is good enough. https://github.com/golangci/golangci-lint/issues/1892

**Comment by Jeffwan** [comment]
There're lots of failures after we enable linter against apiserver and cli.

**Comment by Jeffwan** [comment]
Those issues have been fixed in #158
